sdks/python/apache_beam/transforms/core.py

```diff
@@ -1337,6 +1372,75 @@ def default_type_hints(self):
     return hints
 
 
+class _CombinePerKeyWithHotKeyFanout(PTransform):
+
+  def __init__(self, combine_fn, fanout):
+    self._fanout_fn = (
+        (lambda key: fanout) if isinstance(fanout, int) else fanout)
+    self._combine_fn = combine_fn
+
+  def expand(self, pcoll):
+
+    from apache_beam.transforms.trigger import AccumulationMode
+    combine_fn = self._combine_fn
+    fanout_fn = self._fanout_fn
+
+    class SplitHotCold(DoFn):
+      counter = 0
+
+      def process(self, element):
+        key, value = element
+        fanout = fanout_fn(key)
+        if fanout <= 1:
+          # Boolean indicates this is not an accumulator.
+          yield pvalue.TaggedOutput('cold', (key, (False, value)))
+        else:
+          self.counter += 1  # Round-robin should be more even than random.
```

Comment:
nit; "should be more" more what?

Reply:
More even. But this is ambitious with "even than." I'll reword.